myokit/_model_api.py

```diff
@@ -1722,42 +1724,98 @@ def import_component(self, external_component, new_name=None, var_map=None,
         be raised.
         """
 
-        # Check component is not from this model, is a component, etc.
-        if not isinstance(external_component, myokit.Component):
+        # Check component is list or component and new_name is string or list
+        if isinstance(external_component, list):
```

Comment:
Ideally this would check for the sequence interface instead?
So start with `if isinstance(..., component):` and then have an `else:` either without error checking or with something that catches a type error?

E.g.
```
        if isinstance(external_component, myokit.Component):
            external_component = [external_component]
        else:
            try:
                ok = all(isinstance(c, myokit.Component) for c in external_component)
            except TypeError:
                raise TypeError(
                    'Method import_component() expects a myokit.Component '
                    'or list of myokit.Components'
                )
```